### PR TITLE
fix: make `is_dictionary` optional

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale
We support tag to be dictionary encoding, but in a back-incompatible way, it will throw following errors:

```
err:Arrow field meta key is not found, key:IsDictionary.
```
## Detailed Changes
- When parse arrow meta, make is_dictionary optional

## Test Plan
UT

### CODE REVIEW

File: common_types/src/column_schema.rs [hunk 1]
```diff
@@ -146,6 +146,12 @@ impl ArrowFieldMetaKey {
             ArrowFieldMetaKey::Comment => "field::comment",
         }
     }
+
+    // Only id,is_tag,comment are required fields, other fields should be optional
+    // to keep backward compatible
+    fn is_required(&self) -> bool {
+        matches!(self, Self::Id | Self::IsTag | Self::Comment)
+    }
```
Comment (tanruixiang) [inline common_types/src/column_schema.rs:154]: `MetaKey`

File: common_types/src/column_schema.rs [hunk 2]
```diff
@@ -596,4 +611,46 @@ mod tests {
...
+                    "field::id".to_string() => "1".to_string(),
+                    "field::is_tag".to_string() => "false".to_string(),
+                    "field::comment".to_string() => "abc".to_string(),
+                    "field::is_dictionary".to_string() => "true".to_string()
+                },
+                ArrowFieldMeta {
+                    id: 1,
+                    is_tag: false,
+                    is_dictionary: true,
+                    comment: "abc".to_string(),
```
Comment (chunshao90) [inline common_types/src/column_schema.rs:642]:
```suggestion
                    comment: "abc".to_string(),
                    is_dictionary: true,
```

File: common_types/src/column_schema.rs [hunk 3]
```diff
@@ -596,4 +611,46 @@ mod tests {
...
+                hash_map! {
+                    "field::id".to_string() => "1".to_string(),
+                    "field::is_tag".to_string() => "true".to_string(),
+                    "field::comment".to_string() => "".to_string()
+                },
+                ArrowFieldMeta {
+                    id: 1,
+                    is_tag: true,
+                    is_dictionary: false,
+                    comment: "".to_string(),
```
Comment (chunshao90) [inline common_types/src/column_schema.rs:628]:
```suggestion
                    comment: "".to_string(),
                    is_dictionary: false,
```

**Review by chunshao90 — APPROVED** [review]
LGTM

**Pull request merged**